gigascience/gigadb-website
# Add general accessbile landmarks to layouts

# Pull request for issue: #1361

NOTE: I'm going to recreate this PR with another branch as I've messed up the diff trying to get rid of the formatting :)

This is a pull request for the following functionalities:

- Add header, main and footer landmarks to layouts that missed them
- Add role="search" to existing search forms, so that assistive technologies can identifiy the search form of each page
- Wrap navigation links in <nav> element, and add labels to identify each <nav> (except for the old layouts, which will be eventually replaced)

## How to test?

Open pages using each of the different layouts, run Axe audit (browser extension for local dev https://wave.webaim.org/extension/) and indentify header, main, footer, and nav landmarks. Confirm that search form has role="search" attribute

Layouts:

- new_main: /
- new_faq: /site/faq
- new_dataset: /dataset/100006

## How have functionalities been implemented?

- the different layouts used all accross the site were identified and missing header, footer and main landmarks were added
- for the new layouts navigation links were identified and wrapped in <nav>
- Search forms were found in the codebase and given the role="search" attribute. Relevant search strings: `beginForm('/search`, `<form action="/search/new`

## Any issues with implementation?

Some code changes are due to the autoformatting on save applied by my IDE. The formatting is inconsistent in general. One option would be to apply and enforce some kind of predefined autoformatting / linting. I will attempt to revert the formatting changes after creating this PR.

## Any changes to automated tests?

None

## Any changes to documentation?

None

## Any technical debt repayment?

None

## Any improvements to CI/CD pipeline?

None
